teps/0091-trusted-resources.md

```diff
@@ -313,6 +395,102 @@ What other approaches did you consider and why did you rule them out?  These do
 not need to be as detailed as the proposal, but should include enough
 information to express the idea and why it was not acceptable.
 -->
+### Options to mitigate the possible impact from mutating webhook
+Some possible mitigations/solutions include the following:
+
+1. Verify the Tasks/Pipelines when applying them in the cluster, and update the Resources to add the missing mutating values (e.g. Use `SetDefaults`) everytime we bump the version of Tekton Pipeline.
+2. Avoid mutating Tekton Resources or move the mutating after verification.
+3. Store the encoded resource in annotation before mutating and verify the stored resource at validating webhook.
+4. Do verification in mutating webhook, if failed verification then mark the run as failed and later check and reject the resource in validating webhook.
+
+**Pros vs Cons:**
+| Option | Pros | Cons |
+| -------- | ----------- | ----------- |
+| Call SetDefaults() when signing | Help users add missing values |  Need to update signature when there is update in mutating
+| Skip mutating webhook | Address the mutating issue, can also help to detect malicious mutating webhook| Extra cost for Tekton Pipeline developers
+| Store the object at mutating webhook before mutating| Fix the issue without skipping the mutating webhook |  May not work if the attack happens in mutating webhook and only modify the spec
+| Verify at mutating webhook | No need to skip the mutating or store in annotation | Doesn't work if there is a malicious mutating webhook after the normal mutating webhook
+
+In this TEP we propose to proceed with with option 2.
```

Comment:
SGTM - are we disabling mutating webhooks all the time or only when this feature is enabled?

Reply:
Only when this feature is enabled. And this will only be set for tasks and pipelines not the runs. This is mentioned in the next paragraph